debug/expressions.html

```diff
@@ -0,0 +1,86 @@
+            }
+        }
+    },
+    layers: [
+        {
+            id: 'circle',
+            type: 'circle',
+            source: 'geojson',
+            paint: {
+                'circle-radius': {
+                    type: 'expression',
+                    expression: [
+                        '*',
+                        [ '^', 2, {ref: 'map', key: 'zoom'} ],
+                        0.125,
+                        [
+                            '+',
+                            20,
+                            {ref: 'feature', key: 'x'},
+                            {ref: 'feature', key: 'y'}
+                        ]
+                    ]
```

Comment:
The initialized NSExpression consists of the following object tree:

* NSExpression representing the function `multiply:by:`, the operand `_NSPredicateUtilities`, and the following NSExpressions as arguments:
  * NSExpression representing the function `multiply:by:`, the operand `_NSPredicateUtilities`, and the following NSExpressions as arguments:
    * NSExpression representing the function `raise:toPower:`, the operand `_NSPredicateUtilities`, and the following NSExpressions as arguments:
      * NSExpression representing the key path `map.zoom`
      * NSExpression representing the constant value 2
    * NSExpression representing the constant value 0.125
* etc.

NSExpression comes with a built-in way to evaluate itself given a context dictionary. However, imperative filters aren’t on the roadmap – https://github.com/mapbox/mapbox-gl-native/issues/7860#issuecomment-275257972 – so the iOS/macOS SDK instead translates the AST provided by NSExpression into the AST expected by mbgl.